common/utils-java/src/main/java/org/apache/spark/util/UUIDv7Generator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.util;
+
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Generator for UUIDv7 as defined in RFC 9562.
+ * https://www.rfc-editor.org/rfc/rfc9562.html
+ *
+ * UUIDv7 is a time-ordered UUID that embeds a Unix timestamp in milliseconds.
+ *
+ * The generate() method guarantees monotonicity by tracking the last used timestamp
+ * and increments if necessary when the clock hasn't advanced.
```

Comment:
This severely limits the throughput of the generator, is strict monotonicity really necessary?

Reply:
so this is going to be a stricter version of UUIDv7 implementation, when I suggest moving the code to the common module, I expected that the implementation would be virtually equivalent to JDK 26's one, but it's not. For these cases, please at least assign the method a clear name (as it's exposed publicly), e.g., `generateMonotonic`, and document what the returned value guarantees and what not, in case it may be optimized for performance by changing the underlying implementation in the future.

Reply:
Thanks for the inputs. Considering the trade-offs, I updated to not enforce strict monotonicity in this implementation (and document the limitation) to avoid degrading throughput. JDK26 also does not guarantee monotonicity, but places the responsibility on the caller to provide monotonic timestamps if strict ordering is required.

For Spark `queryId` use case, millisecond-level ordering should be sufficient. If we later need strict monotonicity for specific use cases we could add a separate `generateMonotonic()`.

Reply:
IMHO, monotonicity guarantee is a best practice for UUIDv7, and it is not too hard to implement a lock-free, monotonically increasing, high-throughput UUIDv7 generator. But it should be fine to work on this iteratively.